salt/state.py

```diff
@@ -2591,7 +2586,7 @@ def get_tops(self):
                     tops[saltenv].append({})
                     log.debug('No contents loaded for env: {0}'.format(saltenv))
 
-            if found > 1 and merging_strategy == 'merge':
+            if found > 1 and merging_strategy == 'merge' and not self.opts.get('env_order', None):
```

Comment:
The ``.get()`` is probably unnecessary, since the opts will always contain an ``env_order`` key, and the default value is an empty list. But this is a minor nitpick and it shouldn't prevent this from being merged.